src/sdk/service/network-validation-service/__tests__/network-validation-service.test.unit.ts

```diff
@@ -0,0 +1,391 @@
+import { NetworkValidationService } from '../network-validation-service';
+
+const nvs = NetworkValidationService.getInstance();
+const dummyIpOne = '192.0.0.1';
+const dummyIpTwo = '10.0.0.1';
+const dummyIpThree = '192.168.0.1';
+const dummyIpArrayOne = ['10.0.0.0', '10.0.0.1', '10.0.0.2'];
+const dummyIpArrayTwo = ['10.0.0.11', '10.0.0.12', '10.0.0.13'];
+const dummyIpArrayThree = [
+  '10.0.0.1',
+  '10.0.0.2',
+  '10.0.0.3',
+  '10.0.0.4',
+  '10.0.0.5',
+  '10.0.0.6',
+  '10.0.0.7',
+  '10.0.0.8',
+  '10.0.0.9'
+];
+const dummyGateway = '192.30.250.00';
+const dummyNetmaskOne = '255.255.192.0';
+const dummyNetmaskTwo = '255.255.255.0';
+const dummyNetmaskThree = '255.255.255.255';
+const dummyRangeOne = {
+  start: '10.0.0.1',
+  end: '10.0.0.10'
+};
+const dummyRangeTwo = {
+  start: '10.0.0.10',
+  end: '10.0.0.255'
+};
+
+test('IP Address is valid', () => {
+  expect(nvs.isValid(dummyIpOne)).toBeTruthy();
+});
+
+test('IP Address is valid and not reserved.', () => {
+  expect(nvs.isValidStrict(dummyIpOne)).toBeFalsy();
```

Comment:
Expecting `Falsy` here, so the test description should be updated from `valid` to `invalid`? Same for the rest of the expected `Falsy` tests that are described as `valid`.

Reply:
@SuperConfuserUser I'm testing for both truthiness and falsiness. Plus Cory wanted me to keep the function names the same as the legacy functions.

Reply:
I don't recall asking you to keep the same names. I actually would prefer that you change the names where it makes sense. This case is a good example, I would expect `isValidStrict` and `isValid` to be combined. The name could be changed to isValidIPV4Address and it could take an optional boolean parameter to indicate whether .0 and .255 should be considered valid.

Reply:
And I agree with Chely's comment about the test description not accurately describing what is being tested here

Reply:
Hi Cory. Could you give me an example of what the descriptions should be for some of these tests?

@cfsnyder

Reply:
Actually scratch that. I can probably just look at other tests.